starting_from_scratch_lab/lab 37/lab_37_tar2.js

```diff
@@ -0,0 +1,19 @@
+const cars = [
+  { name: "c1", color: "red", speed: 40 },
+  { name: "c2", color: "blue", speed: 20 },
+];
+
+const fastestCar = getFastestCar(cars);
+console.log(fastestCar.name);
+
+// three solutions
+// one using reduce -> const max = data.reduce((prev, current) => (prev.y > current.y) ? prev : current) (n)
+// second using var maxB = a.sort((a,b)=>b.y-a.y)[0].y;    // 27 chars time complexity:  O(nlogn) (nlogn)
+// third var maxC = Math.max(...a.map(o=>o.y));     // 26 chars time complexity: >O(2n)
+
```

Comment:
Amazing! As an aside there's a popular JavaScript library called lodash, and it provides a function called maxBy that would be really perfect here (although of course one would need to include that library).

It's widely used and worth taking a look at:
https://lodash.com/docs/#maxBy